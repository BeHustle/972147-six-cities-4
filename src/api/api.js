import axios from 'axios';

const URL = `https://4.react.pages.academy/six-cities`;
const TIMEOUT = 5000;

export const createAPI = (dispatch = null) => {
  const api = axios.create({
    baseURL: URL,
    timeout: TIMEOUT,
    withCredentials: true
  });

  const onSuccess = (response) => response;
  const onFail = ({response}) => {
    dispatch();
    throw response;
  };

  api.interceptors.response.use(onSuccess, onFail);
  return api;
};
