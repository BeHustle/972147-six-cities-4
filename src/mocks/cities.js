export const cities = [
  {
    id: 1,
    name: `Amsterdam`,
    coordinates: [52.38333, 4.9]
  },
  {
    id: 2,
    name: `Paris`,
    coordinates: [52.38333, 4.9]
  },
  {
    id: 3,
    name: `Cologne`,
    coordinates: [52.38333, 4.9]
  },
  {
    id: 4,
    name: `Brussels`,
    coordinates: [52.38333, 4.9]
  },
  {
    id: 5,
    name: `Hamburg`,
    coordinates: [52.38333, 4.9]
  },
  {
    id: 6,
    name: `Dusseldorf`,
    coordinates: [52.38333, 4.9]
  },
];
