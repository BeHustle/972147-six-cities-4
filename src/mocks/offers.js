export const offers = [
  {
    id: 1,
    name: `Beautiful & luxurious apartment at great location`,
    price: 132,
    image: `img/apartment-01.jpg`,
    type: `Private room`,
    isPremium: false,
    inBookmarks: true,
    rating: 4
  },
  {
    id: 2,
    name: `Wood and stone place`,
    price: 254,
    image: `img/apartment-02.jpg`,
    type: `Apartment`,
    isPremium: false,
    inBookmarks: false,
    rating: 5
  },
  {
    id: 3,
    name: `Canal View Prinsengracht`,
    price: 178,
    image: `img/apartment-03.jpg`,
    type: `Private room`,
    isPremium: true,
    inBookmarks: true,
    rating: 3
  },
  {
    id: 4,
    name: `Nice, cozy, warm big bed apartment`,
    price: 995,
    image: `img/room.jpg`,
    type: `Apartment`,
    isPremium: true,
    inBookmarks: false,
    rating: 1
  },
  {
    id: 5,
    name: `Beautiful & luxurious apartment at great location`,
    price: 80,
    image: `img/apartment-01.jpg`,
    type: `Private room`,
    isPremium: false,
    inBookmarks: true,
    rating: 2
  },
];
