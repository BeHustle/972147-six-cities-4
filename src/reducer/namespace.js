const Namespace = {
  DATA: `DATA`,
  APP: `APP`,
  USER: `USER`
};

export {Namespace};
