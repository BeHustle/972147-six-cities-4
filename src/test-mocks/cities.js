export const cities = [
  {
    id: 1,
    name: `Amsterdam`,
    coordinates: [52.38333, 4.9],
    zoom: 10
  },
  {
    id: 2,
    name: `Paris`,
    coordinates: [48.85875194, 2.33926391],
    zoom: 10
  },
  {
    id: 3,
    name: `Cologne`,
    coordinates: [50.938361, 6.959974],
    zoom: 10
  },
  {
    id: 4,
    name: `Brussels`,
    coordinates: [50.8465573, 4.351697],
    zoom: 10
  },
  {
    id: 5,
    name: `Hamburg`,
    coordinates: [53.550341, 10.000654],
    zoom: 10
  },
  {
    id: 6,
    name: `Dusseldorf`,
    coordinates: [51.2254018, 6.7763137],
    zoom: 10
  },
];
