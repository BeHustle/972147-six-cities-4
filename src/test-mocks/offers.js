export const offers = [
  {
    id: 1,
    name: `Beautiful & luxurious apartment at great locatccion`,
    price: 132,
    image: `img/apartment-01.jpg`,
    images: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
    ],
    type: `hotel`,
    isPremium: true,
    inBookmarks: true,
    rating: 4.1,
    rooms: `3 Bedrooms`,
    guests: `Max 4 adults`,
    facilities: [`Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    author: {
      name: `Angelina`,
      avatar: `img/avatar-angelina.jpg`,
      isSuper: true,
    },
    text: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the
      bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    coordinates: [52.3909553943508, 4.85309666406198],
    cityId: 1,
  },
  {
    id: 2,
    name: `Wood and stone place`,
    price: 254,
    image: `img/apartment-02.jpg`,
    images: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
    ],
    type: `apartment`,
    isPremium: false,
    inBookmarks: false,
    coordinates: [52.369553943508, 4.85309666406198],
    rating: 5,
    rooms: `1 Bedrooms`,
    guests: `Max 2 adults`,
    facilities: [`Wi-Fi`,
      `Washing machine`,
      `Towels`,
    ],
    author: {
      name: `Max`,
      avatar: `img/avatar-max.jpg`,
      isSuper: true,
    },
    text: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the
      bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    cityId: 1,
  },
  {
    id: 3,
    name: `Canal View Prinsengracht`,
    price: 178,
    image: `img/apartment-03.jpg`,
    images: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
    ],
    type: `room`,
    isPremium: true,
    inBookmarks: true,
    rating: 3,
    coordinates: [48.85875194, 2.33926391],
    rooms: `2 Bedrooms`,
    guests: `Max 2 adults and 1 children`,
    facilities: [`Wi-Fi`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
    ],
    author: {
      name: `Alex`,
      avatar: ``,
      isSuper: true,
    },
    text: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the
      bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    cityId: 2,
  },
  {
    id: 4,
    name: `Nice, cozy, warm big bed apartment`,
    price: 995,
    image: `img/room.jpg`,
    images: [
      `img/room.jpg`,
      `img/apartment-01.jpg`,
      `img/apartment-02.jpg`,
      `img/apartment-03.jpg`,
      `img/studio-01.jpg`,
      `img/apartment-01.jpg`,
    ],
    type: `house`,
    isPremium: true,
    inBookmarks: false,
    coordinates: [50.938361, 6.959974],
    rating: 1,
    rooms: `5 Bedrooms`,
    guests: `Max 3 adults and 2 children`,
    facilities: [`Wi-Fi`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`,
    ],
    author: {
      name: `Steven`,
      avatar: ``,
      isSuper: true,
    },
    text: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
      The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the
      bustle of the city comes to rest in this alley flowery and colorful.`,
    ],
    cityId: 3,
  }];

export const serverOffers = [
  {
    "bedrooms": 3,
    "city": {
      "location": {
        "latitude": 52.370216,
        "longitude": 4.895168,
        "zoom": 10
      },
      "name": `Amsterdam`
    },
    "description": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    "goods": [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    "host": {
      "avatar_url": `img/1.png`,
      "id": 3,
      "is_pro": true,
      "name": `Angelina`
    },
    "id": 1,
    "images": [`img/1.png`, `img/2.png`],
    "is_favorite": false,
    "is_premium": false,
    "location": {
      "latitude": 52.35514938496378,
      "longitude": 4.673877537499948,
      "zoom": 8
    },
    "max_adults": 4,
    "preview_image": `img/1.png`,
    "price": 120,
    "rating": 4.8,
    "title": `Beautiful & luxurious studio at great location`,
    "type": `apartment`
  }
];
