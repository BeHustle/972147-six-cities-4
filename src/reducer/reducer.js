import {combineReducers} from 'redux';
import {Namespace} from './namespace';
import {reducer as app} from './app/app.reducer';
import {reducer as data} from './data/data.reducer';
import {reducer as user} from './user/user.reducer';

export default combineReducers({
  [Namespace.APP]: app,
  [Namespace.DATA]: data,
  [Namespace.USER]: user
});
