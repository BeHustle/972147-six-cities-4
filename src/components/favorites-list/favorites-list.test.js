import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import {applyMiddleware, createStore} from 'redux';
import {composeWithDevTools} from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import {createAPI} from '../../api/api.js';
import reducer from '../../reducer/reducer.js';
import {cities} from '../../test-mocks/cities.js';
import {offers} from '../../test-mocks/offers.js';
import FavoritesList from './favorites-list.jsx';

const api = createAPI();

const store = createStore(
    reducer,
    composeWithDevTools(
        applyMiddleware(thunk.withExtraArgument(api))
    )
);

it(`Render empty favorites list`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <FavoritesList
            offers={offers}
            cities={cities}
            onFavoriteClick={() => {}}
            onCardTitleClick={() => {}}/>
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
