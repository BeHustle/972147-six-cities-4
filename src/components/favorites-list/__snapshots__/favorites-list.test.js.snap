// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render empty favorites list 1`] = `
<ul
  className="favorites__list"
>
  <li
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="#"
        >
          <span>
            Amsterdam
          </span>
        </a>
      </div>
    </div>
    <div
      className="favorites__places"
    >
      <article
        className="place-card favorites__card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="place-card__image-wrapper favorites__image-wrapper"
        >
          <a
            href="#"
          >
            <img
              alt="Place image"
              className="place-card__image"
              height="200"
              src="img/apartment-01.jpg"
              width="260"
            />
          </a>
        </div>
        <div
          className="place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €
                132
              </b>
              <span
                className="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              className="place-card__bookmark-button button place-card__bookmark-button--active"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="place-card__rating rating"
          >
            <div
              className="place-card__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "100%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            className="place-card__name"
          >
            <a
              href="#"
              onClick={[Function]}
            >
              Beautiful & luxurious apartment at great locatccion
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            hotel
          </p>
        </div>
      </article>
      <article
        className="place-card favorites__card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="place-card__image-wrapper favorites__image-wrapper"
        >
          <a
            href="#"
          >
            <img
              alt="Place image"
              className="place-card__image"
              height="200"
              src="img/apartment-02.jpg"
              width="260"
            />
          </a>
        </div>
        <div
          className="place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €
                254
              </b>
              <span
                className="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              className="place-card__bookmark-button button false"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="place-card__rating rating"
          >
            <div
              className="place-card__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "100%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            className="place-card__name"
          >
            <a
              href="#"
              onClick={[Function]}
            >
              Wood and stone place
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            apartment
          </p>
        </div>
      </article>
    </div>
  </li>
  <li
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="#"
        >
          <span>
            Paris
          </span>
        </a>
      </div>
    </div>
    <div
      className="favorites__places"
    >
      <article
        className="place-card favorites__card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="place-card__image-wrapper favorites__image-wrapper"
        >
          <a
            href="#"
          >
            <img
              alt="Place image"
              className="place-card__image"
              height="200"
              src="img/apartment-03.jpg"
              width="260"
            />
          </a>
        </div>
        <div
          className="place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €
                178
              </b>
              <span
                className="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              className="place-card__bookmark-button button place-card__bookmark-button--active"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="place-card__rating rating"
          >
            <div
              className="place-card__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "60%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            className="place-card__name"
          >
            <a
              href="#"
              onClick={[Function]}
            >
              Canal View Prinsengracht
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            room
          </p>
        </div>
      </article>
    </div>
  </li>
  <li
    className="favorites__locations-items"
  >
    <div
      className="favorites__locations locations locations--current"
    >
      <div
        className="locations__item"
      >
        <a
          className="locations__item-link"
          href="#"
        >
          <span>
            Cologne
          </span>
        </a>
      </div>
    </div>
    <div
      className="favorites__places"
    >
      <article
        className="place-card favorites__card"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <div
          className="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="place-card__image-wrapper favorites__image-wrapper"
        >
          <a
            href="#"
          >
            <img
              alt="Place image"
              className="place-card__image"
              height="200"
              src="img/room.jpg"
              width="260"
            />
          </a>
        </div>
        <div
          className="place-card__info"
        >
          <div
            className="place-card__price-wrapper"
          >
            <div
              className="place-card__price"
            >
              <b
                className="place-card__price-value"
              >
                €
                995
              </b>
              <span
                className="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              className="place-card__bookmark-button button false"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="place-card__rating rating"
          >
            <div
              className="place-card__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "20%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            className="place-card__name"
          >
            <a
              href="#"
              onClick={[Function]}
            >
              Nice, cozy, warm big bed apartment
            </a>
          </h2>
          <p
            className="place-card__type"
          >
            house
          </p>
        </div>
      </article>
    </div>
  </li>
  
  
  
</ul>
`;
